### PR TITLE
feat(cloudtrail): custom log delivery role and sendToCloudWatchLogs on intent-based

### LINKED ISSUE #9583: [aws-cloudtrail] Custom log delivery role
Hello. Can you add a CloudTrail Trail prop for passing in the role to use for CloudWatch logs delivery ? As it is, CDK creates the role for us.

### Use Case

To support our permission strucutre, we namespace all the service roles we pass to AWS services in a particular way. The role name that CDK chooses on our behalf doesn't satisfy our naming convention.

### Proposed Solution

Add an optional `role: IRole` prop to the Trail constructor. If present, have the trail use this role for cloudwatch log delivery (if enabled)

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by 14kw**
- Enable custom log delivery role
- Enable sendToCloudWatchLogs on intent-based
  - https://github.com/aws/aws-cdk/pull/11104#discussion_r522425586

Even if sendToCloudWatchLogs is undefined, it will be evaluated if the properties associated with CloudWatchLogs are set.

closes #9583

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by NetaNir** [comment]
We have reviewed this pull request and marking it as p2, which means that we are unable to work on it immediately. We especially value community contributed code, so this is not a final decision.
We use +1s to help prioritize our work, and are happy to revaluate this pull request based on community feedback. You can reach out to the cdk.dev community on Slack to solicit support for reprioritization.

File: packages/@aws-cdk/aws-cloudtrail/test/cloudtrail.test.ts [hunk 1]
```diff
@@ -255,7 +255,6 @@ describe('cloudtrail', () => {
       test('enabled and custom retention', () => {
         const stack = getTestStack();
         new Trail(stack, 'MyAmazingCloudTrail', {
-          sendToCloudWatchLogs: true,
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-cloudtrail/test/cloudtrail.test.ts:258]: While we should have at least one test case where this is removed and the other relevant fields are set, making this true, we should also still have a test case where this is set explicitly. Overall, I'd like to see a bit more test coverage for the various cases with this new behavior (including throwing an error as suggested above).

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution. Overall this PR is great! I think this is a valuable change, just a couple comments below, but I'd love to see this get merged.

**Comment by TheRealAmazonKendra** [comment]
This PR has been deemed to be abandoned, and will be automatically closed. Please create a new PR for these changes if you think this decision has been made in error.

**Pull request closed without merging**